charts/nr-k8s-otel-collector/templates/deployment-configmap.yaml

```diff
@@ -0,0 +1,383 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "newrelic.common.naming.fullname" . }}-deployment-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+data:
+  deployment-config.yaml: |
+    receivers:
+      otlp:
+        protocols:
+          {{ if .Values.enableGrpcGatewayEndpoint }}
+          grpc:
+            endpoint: ${env:MY_POD_IP}:4317
+          {{ end }}
```

Comment:
I would remove all extra functionalities that are not required for now. Otherwise, you will need to support them forever.

(this unless it is a requirement from product 😄 )

Reply:
If there is a follow up PR to improve the configuration part, I would left this for that PR.

There are still lots of things that are not implemented (like the `LowDataMode`) and too many configurations that need to change depending if they are running on `hostNetwork` mode (for example).

These kind of toogles that are not in a tree (`.Values.enableGrpcGatewayEndpoint` instead of `.Values.gateways.grpc.enable`) are an anti-pattern and prone to be HARD to maintain.

Example:
https://github.com/newrelic/nri-kubernetes/blob/a4bdffd9d96b446c4fa7ba8351f981656b3b091d/charts/newrelic-infrastructure/templates/_helpers_compatibility.tpl#L10-L30
That is used here:
https://github.com/newrelic/nri-kubernetes/blob/a4bdffd9d96b446c4fa7ba8351f981656b3b091d/charts/newrelic-infrastructure/templates/ksm/scraper-configmap.yaml#L14

Reply:
Renamed to `.Values.gateway.grpc.enable`